service/replication.go

```diff
@@ -214,8 +214,19 @@ func (s *service) CreateRemoteVolume(ctx context.Context, req *replication.Creat
 
 // DeleteLocalVolume deletes the backend volume on the storage array.
 func (s *service) DeleteLocalVolume(ctx context.Context, req *replication.DeleteLocalVolumeRequest) (*replication.DeleteLocalVolumeResponse, error) {
+	Log.Printf("[DeleteLocalVolume] - req %+v", req)
 
-	Log.Error("DeleteLocalVolume is not yet implemented")
+	volHandleCtx := req.GetVolumeHandle()
+
+	if volHandleCtx == "" {
+		return nil, status.Error(codes.InvalidArgument, "volume handle is required")
+	}
+
+	_, err := s.DeleteVolume(ctx, &csi.DeleteVolumeRequest{VolumeId: volHandleCtx})
```

Comment:
What will happen if the replicated target volume is still part of replication pair/group when this deleteVolume() call is made? I think it is better to check those and fail, and proceed to DeleteVolume() during retry when those checks pass.

Reply:
This situation is already handled in the `DeleteVolume` function (https://github.com/dell/csi-powerflex/blob/main/service/controller.go#L623).

We check to see if the volume is part of a pair and delete the pair before deleting the volume.

Reply:
That is the current flow at the moment. When the source PV is deleted, it will remove the replication pair. The DeleteLocalVolume will then be executed which at this point the volume would have been already removed from the replication pair and it just finishes the clean up of the remaining backend volume.

We could check to see if the volume is marked for replication before calling this DeleteVolume but it should have already been handled with the deletion flow of the source PV.

Thoughts?

Reply:
Added check of volume replication state.